sky/backends/backend_utils.py

```diff
@@ -276,21 +276,36 @@ def _optimize_file_mounts(yaml_path: str) -> None:
 
 
 def path_size_megabytes(path: str) -> int:
-    """Returns the size of 'path' (directory or file) in megabytes."""
+    """Returns the size of 'path' (directory or file) in megabytes.
+
+    Returns:
+        If successful: the size of 'path' in megabytes, rounded down. Otherwise,
+        -1.
+    """
     resolved_path = pathlib.Path(path).expanduser().resolve()
     git_exclude_filter = ''
     if (resolved_path / command_runner.GIT_EXCLUDE).exists():
         # Ensure file exists; otherwise, rsync will error out.
         git_exclude_filter = command_runner.RSYNC_EXCLUDE_OPTION.format(
             str(resolved_path / command_runner.GIT_EXCLUDE))
-    rsync_output = str(
-        subprocess.check_output(
-            f'rsync {command_runner.RSYNC_DISPLAY_OPTION} '
-            f'{command_runner.RSYNC_FILTER_OPTION} '
-            f'{git_exclude_filter} --dry-run {path!r}',
-            shell=True).splitlines()[-1])
-    total_bytes = rsync_output.split(' ')[3].replace(',', '')
-    return int(float(total_bytes)) // 10**6
+    rsync_output = ''
+    try:
+        rsync_output = str(
+            subprocess.check_output(
+                f'rsync {command_runner.RSYNC_DISPLAY_OPTION} '
+                f'{command_runner.RSYNC_FILTER_OPTION} '
+                f'{git_exclude_filter} --dry-run {path!r}',
+                shell=True))
+    except subprocess.CalledProcessError:
+        return -1
+    match = re.search(r'total size is (\d+)', rsync_output)
```

Comment:
Question: It seems there will be `,` separating the numbers, will `\d+` recognize it?
```
total size is 250,957,728  speedup is 330.19 (DRY RUN)
```

Reply:
My rsync:
```
» rsync --version                          
rsync  version 2.6.9  protocol version 29
```
What's your version?

Reply:
Mine is 
```
rsync --version                                                                                      
rsync  version 3.2.3  protocol version 31
```

I think I upgraded the rsync with homebrew once.

Reply:
Good catch! Changed the pattern.

I did not proceed to test more versions as this is used for warning only & can fix on-demand.